src/data/adHocFilter.ts

```diff
@@ -0,0 +1,20 @@
+
+export function applyAdHocFilter(rawSql: string, adHocFilters: AdHocVariableFilter[]): string {
+  if (rawSql != '' && adHocFilters && Object.keys(adHocFilters).length > 0) {
+    let whereClause = 'WHERE';
+    for (let k of adHocFilters) {
+      let v = isNaN(Number(k.value)) ? `'${k.value}'` : Number(k.value)
+      whereClause += ` ${k.key} ${k.operator} ${v} ${k.condition ? k.condition : 'AND'}`;
```

Comment:
if there is no condition will this apply an unnecessary AND?  seems like the last filter should not apply a condition

if the query already has a where clause this probably doesn't work?

Reply:
Oh wait! What if we apply adhoc filters to inline queries. I am not sure if there would ever be a case where you are wanting an adhoc filter to apply to your outer query if there is an inline one. @scottlepp what do you think?

Reply:
I was thinking we could just apply it to the outer query.  I'm guessing ad hoc filters won't be used with complex nested queries.  If the outer query ( or any simple query ) already has a where clause, we would just append to that.